pyhms/core/initializers.py

```diff
@@ -0,0 +1,122 @@
+        super().__init__(bounds)
+        self.sampler = LatinHypercube(d=len(bounds), seed=random_seed)
+
+    def sample_pop(self, pop_size: int) -> np.ndarray:
+        sample = self.sampler.random(pop_size)
+        return linear_scale(sample, self._bounds)
+
+
+class SobolGlobalInitializer(PopInitializer):
+    def __init__(self, bounds: np.ndarray, random_seed: int = None):
+        super().__init__(bounds)
+        self.sampler = Sobol(d=len(bounds), scramble=True, seed=random_seed)
+
+    def sample_pop(self, pop_size: int) -> np.ndarray:
+        sample = self.sampler.random(pop_size)
+        return linear_scale(sample, self._bounds)
+
+
+class GaussianInitializer(PopInitializer):
+    def __init__(self, seed: np.ndarray, std_dev: float, bounds: np.ndarray | None = None):
+        super().__init__(bounds)
+        self.sampler = sample_normal(seed, std_dev, bounds)
+        self._seed = seed
+
+    def sample_pop(self, pop_size: int) -> np.ndarray:
+        return self.sampler(pop_size)
+
+
+class GaussianInitializerWithSeedInject(SeededPopInitializer):
+    def __init__(
+        self, seed: Individual, std_dev: float, bounds: np.ndarray | None = None, preserve_fitness: bool = True
+    ):
+        super().__init__(bounds)
+        self.sampler = sample_normal(seed.genome, std_dev, bounds)
+        self._preserve_fitness = preserve_fitness
+        self._seed_ind = seed
+
+    def get_seed(self, problem: Problem) -> Individual:
+        if self._preserve_fitness:
+            return Individual(genome=self._seed_ind.genome, problem=problem, fitness=self._seed_ind.fitness)
+        else:
+            return Individual(genome=self._seed_ind.genome, problem=problem)
+
+    def sample_pop(self, pop_size: int) -> np.ndarray:
+        return self.sampler(pop_size - 1)
+
+    def __call__(self, pop_size: int | None, problem: Problem):
+        return super().__call__(pop_size, problem) + [self.get_seed(problem)]
+
+
+class InjectionInitializer(SeededPopInitializer):
+    def __init__(
+        self, injected_population: list[Individual], bounds: np.ndarray | None = None, preserve_fitness: bool = True
+    ):
+        super().__init__(bounds)
+        self._preserve_fitness = preserve_fitness
+        self.injected_population = injected_population
+
+    def sample_pop(self, _: int | None) -> np.ndarray:
+        raise NotImplementedError("InjectionInitializer does not have a sampler. Use __call__ instead.")
+
+    def __call__(self, pop_size: int | None, problem: Problem) -> list[Individual]:
+        if self._preserve_fitness:
+            to_inject = [
+                Individual(genome=ind.genome, problem=problem, fitness=ind.fitness) for ind in self.injected_population
+            ]
+        else:
+            to_inject = [Individual(genome=ind.genome, problem=problem) for ind in self.injected_population]
+
+        if pop_size is None or pop_size >= len(self.injected_population):
```

Comment:
I assume that sampled population has to be equal to `pop_size`. Why do we want to return `self.injected_population` if its size doesn't match `pop_size`?

Reply:
I imagined a scenario, when we do have some evaluated individuals yet less, than required population size. In that case, we would want to use the injected population and sample some additional individuals.

Reply:
ok, it makes sense